app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -83,7 +87,9 @@ class SubmissionFileManager extends React.Component {
         delete_files: fileKeys,
         grouping_id: this.props.grouping_id
       }
-    }).then(typeof this.props.onChange === 'function' ? this.props.onChange : this.fetchData)
+    }).then(typeof this.props.onChange === 'function' ?
+      this.setState({viewFile: null, viewFileType: null, viewFileURL: null}, () => {this.props.onChange;}) :
```

Comment:
This part isn't quite right. You either need to call `this.props.onChange()`, or (better) simply pass `this.props.onChange` without wrapping it in a thunk.